tests/link/pytorch/test_slinalg.py

```diff
@@ -0,0 +1,141 @@
+
+    with pytest.raises(NotImplementedError):
+        compare_pytorch_and_py(
+            out_fg,
+            [
+                np.array(
+                    [[6, 3, 1, 5], [3, 0, 5, 1], [1, 5, 6, 2], [5, 1, 2, 2]]
+                ).astype(config.floatX),
+                np.array(
+                    [[10, 0, 1, 3], [0, 12, 7, 8], [1, 7, 14, 2], [3, 8, 2, 16]]
+                ).astype(config.floatX),
+            ],
+        )
+    compare_pytorch_and_py(
+        out_fg,
+        [
+            np.array([[6, 3, 1, 5], [3, 0, 5, 1], [1, 5, 6, 2], [5, 1, 2, 2]]).astype(
+                config.floatX
+            ),
+            None,
+        ],
+    )
+
+
+def test_pytorch_basic():
+    rng = np.random.default_rng(28494)
+
+    x = matrix("x")
+    b = vector("b")
+
+    out = pt_slinalg.cholesky(x)
+    out_fg = FunctionGraph([x], [out])
+    compare_pytorch_and_py(
+        out_fg,
+        [
+            (np.eye(10) + rng.standard_normal(size=(10, 10)) * 0.01).astype(
+                config.floatX
+            )
+        ],
+    )
+
+    out = pt_slinalg.cholesky(x, lower=False)
+    out_fg = FunctionGraph([x], [out])
+    compare_pytorch_and_py(
+        out_fg,
+        [
+            (np.eye(10) + rng.standard_normal(size=(10, 10)) * 0.01).astype(
+                config.floatX
+            )
+        ],
+    )
+
+    out = pt_slinalg.solve(x, b)
+    out_fg = FunctionGraph([x, b], [out])
+    compare_pytorch_and_py(
+        out_fg,
+        [
+            np.eye(10).astype(config.floatX),
+            np.arange(10).astype(config.floatX),
+        ],
+    )
+
+
+@pytest.mark.parametrize(
+    "check_finite",
+    (False, pytest.param(True, marks=pytest.mark.xfail(raises=NotImplementedError))),
+)
+@pytest.mark.parametrize("lower", [False, True])
+@pytest.mark.parametrize("trans", [0, 1, 2, "S"])
+def test_pytorch_SolveTriangular(trans, lower, check_finite):
+    x = matrix("x")
+    b = vector("b")
+
+    out = pt_slinalg.solve_triangular(
+        x,
+        b,
+        trans=trans,
+        lower=lower,
+        check_finite=check_finite,
+    )
+    out_fg = FunctionGraph([x, b], [out])
+    compare_pytorch_and_py(
+        out_fg,
+        [
+            np.eye(10).astype(config.floatX),
+            np.arange(10).astype(config.floatX),
+        ],
+    )
+
+
+def test_pytorch_block_diag():
+    A = matrix("A")
+    B = matrix("B")
+    C = matrix("C")
+    D = matrix("D")
+
+    out = pt_slinalg.block_diag(A, B, C, D)
+    out_fg = FunctionGraph([A, B, C, D], [out])
+
+    compare_pytorch_and_py(
+        out_fg,
+        [
+            np.random.normal(size=(5, 5)).astype(config.floatX),
+            np.random.normal(size=(3, 3)).astype(config.floatX),
+            np.random.normal(size=(2, 2)).astype(config.floatX),
+            np.random.normal(size=(4, 4)).astype(config.floatX),
+        ],
+    )
+
+
+@pytest.mark.xfail(reason="Blockwise not implemented")
+def test_pytorch_block_diag_blockwise():
```

Comment:
I don't think we need this if it just xfails. You don't blockwise test any of the other linalg ops?

Reply:
I think in this case the `Blockwise` is not removed after compilation. 

```
Blockwise{BlockDiagonal{n_inputs=2}, (m0,n0),(m1,n1)->(m,n)} [id A] <Tensor3(float64, shape=(?, ?, ?))> 0
 ├─ A [id B] <Tensor3(float64, shape=(?, ?, ?))>
 └─ B [id C] <Tensor3(float64, shape=(?, ?, ?))>
```